firedrake/bcs.py

```diff
@@ -57,7 +58,7 @@ def __init__(self, V, g, sub_domain, method="topological"):
         self._original_arg = self.function_arg
         if sub_domain == "on_boundary":
             self.sub_domain = \
-                map(int, V.mesh().topology.exterior_facets.unique_markers)
+                list(map(int, V.mesh().topology.exterior_facets.unique_markers))
```

Comment:
Sure, I was just noting that we'd presumably prefer if the `sub_domain` property were immutable.